bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/gcov/GcovFileView.java

```diff
@@ -0,0 +1,379 @@
+{
+
+	public static final String ID = "com.espressif.idf.ui.gcov.reportsView";
+
+	private Table table;
+	private IProject selectedProject;
+
+	@Override
+	public void createPartControl(Composite parent)
+	{
+		table = new Table(parent, SWT.MULTI | SWT.BORDER | SWT.FULL_SELECTION);
+		table.setLinesVisible(true);
+		table.setHeaderVisible(true);
+
+		String[] titles = { Messages.TableCol_FileName, Messages.TableCol_Path, Messages.TableCol_LastModifiedGCNO,
+				Messages.TableCol_LastModifiedGCDA, Messages.TableCol_SizeGCNO, Messages.TableCol_SizeGCDA };
+		for (int i = 0; i < titles.length; i++)
+		{
+			TableColumn column = new TableColumn(table, SWT.NONE);
+			column.setWidth(180);
+			column.setText(titles[i]);
+			column.addSelectionListener(new TableColumnSelectionAdapter(column, i));
+		}
+
+		table.addListener(SWT.MouseDoubleClick, new Listener()
+		{
+			@Override
+			public void handleEvent(Event event)
+			{
+				Point pt = new Point(event.x, event.y);
+				TableItem item = table.getItem(pt);
+				IFile file = (IFile) item.getData();
+
+				if (item != null && file != null)
+				{
+					try
+					{
+						GcovUtility.setUpDialog(file, IDFUtil.getELFFilePath(selectedProject).toOSString());
+						IDE.openEditor(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage(), file);
+					}
+					catch (PartInitException e)
+					{
+						e.printStackTrace();
+					}
+				}
+			}
+		});
+
+		// Create Toolbar and Buttons
+		IToolBarManager mgr = getViewSite().getActionBars().getToolBarManager();
+		Action refreshAction = new Action("Refresh")
+		{
+			public void run()
+			{
+				refreshList();
+			}
+		};
+		mgr.add(refreshAction);
+
+		// Initial population of the list
+		refreshList();
+	}
+
+	@Override
+	public void selectionChanged(IWorkbenchPart part, ISelection selection)
+	{
+		if (!(selection instanceof IStructuredSelection))
+			return;
+		IStructuredSelection ssel = (IStructuredSelection) selection;
+		Object obj = ssel.getFirstElement();
+		if (obj instanceof IProject)
+		{
+			setSelectedProject((IProject) obj);
+		}
+	}
+
+	private void refreshList()
+	{
+		for (TableItem item : table.getItems())
+		{
+			item.dispose();
+		}
+
+		if (getSelectedProject() == null)
+		{
+			// Get the current selection
+			ISelectionService selectionService = PlatformUI.getWorkbench().getActiveWorkbenchWindow()
+					.getSelectionService();
+			IStructuredSelection structuredSelection = (IStructuredSelection) selectionService
+					.getSelection("org.eclipse.ui.navigator.ProjectExplorer"); //$NON-NLS-1$
+
+			if (structuredSelection != null && !structuredSelection.isEmpty())
+			{
+				Object firstElement = structuredSelection.getFirstElement();
+				if (firstElement instanceof IProject)
+				{
+					setSelectedProject((IProject) firstElement);
+				}
+			}
+
+			// Now we check the GCOV Utility to see if any project was selected
+			if (getSelectedProject() == null)
+			{
+				setSelectedProject(GcovUtility.getSelectedProject());
+			}
```

Comment:
Is the Utility class really necessary for this? I believe we can retrieve a mapped project from the active launch configuration at this moment, can't we?

Reply:
I did this because we may have multiple projects and user can right click on the project and launch this view